core/templates/pages/skill-editor-page/skill-preview-tab/skill-preview-tab.component.ts

```diff
@@ -0,0 +1,135 @@
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for the skill preview tab.
+ */
+
+require(
+  'pages/exploration-player-page/learner-experience/' +
+    'conversation-skin.directive.ts');
+require('pages/skill-editor-page/services/skill-editor-state.service.ts');
+require('pages/review-test-page/review-test-engine.service.ts');
+require('domain/state_card/StateCardObjectFactory.ts');
+require(
+  'pages/exploration-player-page/learner-experience/tutor-card.directive.ts');
+require(
+  'components/question-directives/question-player/' +
+    'question-player.directive.ts');
+require('pages/review-test-page/review-test-page.constants.ajs.ts');
+require('pages/review-test-page/review-test-engine.service.ts');
+require('domain/question/editable-question-backend-api.service.ts');
+require('domain/question/QuestionObjectFactory.ts');
+require(
+  'pages/exploration-player-page/services/question-player-engine.service.ts');
+require('services/contextual/url.service.ts');
+require('services/context.service.ts');
+require(
+  'domain/topics_and_skills_dashboard/' +
+    'topics-and-skills-dashboard-backend-api.service.ts');
+
+angular.module('oppia').component('skillPreviewTab', {
+  template: require('./skill-preview-tab.component.html'),
+  controllerAs: '$ctrl',
+  controller: [
+    '$scope', 'ContextService', 'QuestionBackendApiService',
+    'QuestionObjectFactory', 'QuestionPlayerEngineService',
+    'SkillEditorStateService', 'StateCardObjectFactory', 'UrlService',
+    function(
+        $scope, ContextService, QuestionBackendApiService,
+        QuestionObjectFactory, QuestionPlayerEngineService,
+        SkillEditorStateService, StateCardObjectFactory, UrlService) {
+      var ctrl = this;
+      const INTERACTION_TYPES = {
+        ALL: 'All',
+        TEXT_INPUT: 'Text Input',
+        MULTIPLE_CHOICE: 'Multiple Choice',
+        NUMERIC_INPUT: 'Numeric Input',
+        ITEM_SELECTION: 'Item Selection'
+      };
+      ctrl.$onInit = function() {
+        ctrl.skillId = UrlService.getSkillIdFromUrl();
+        SkillEditorStateService.loadSkill(ctrl.skillId);
+        ctrl.questionTextFilter = '';
+        ctrl.interactionFilter = INTERACTION_TYPES.ALL;
+        ctrl.displayCardIsInitialized = false;
+        ctrl.questionsFetched = false;
+        ctrl.ALLOWED_QUESTION_INTERACTIONS = [];
+        for (let interaction in INTERACTION_TYPES) {
+          ctrl.ALLOWED_QUESTION_INTERACTIONS.push(
+            INTERACTION_TYPES[interaction]);
+        }
+        ctrl.skill = SkillEditorStateService.getSkill();
+        ctrl.htmlData = ctrl.skill.getConceptCard().getExplanation().getHtml();
+
+        QuestionBackendApiService.fetchQuestions(
+          [ctrl.skillId], 20, false).then((response) => {
```

Comment:
Avoid magic numbers, move 20 to a constant.

Reply:
```suggestion
          [ctrl.skillId], 20, false).then(response => {
```

Reply:
Done

Reply:
`..).then(response => {` is not done.